### PR TITLE
Improve docs on Batch tasks

### LINKED ISSUE #741: Improve documentation on Batch tasks.
- [ ] Document `QuEraTaskStatus` Enum and link that docs in `RemoteBatch` documentation. 
- [ ] Improve documentation on `LocalBatch` and `RemoteBatch`

bounty: 20 USD

### PR DESCRIPTION
**Pull request by WingCode**
closes #741

### CODE REVIEW

File: src/bloqade/submission/ir/task_results.py [hunk 1]
```diff
@@ -16,6 +16,23 @@ class QuEraShotStatusCode(str, Enum):
...
+    - Running: The task is currently running.
+    - Completed: The task has completed successfully.
+    - Failed: The task has failed.
+    - Cancelled: The task has been cancelled.
+    - Executing: The task is currently being executed.
+    - Enqueued: The task is in the queue waiting to be executed.
+    - Accepted: The task has been accepted for execution.
+    - Unaccepted: The task has not been accepted for execution.
+    - Partial: The task has partially completed.
+    - Unsubmitted: The task has not been submitted for execution.
```
Comment (Roger-luo) [inline src/bloqade/submission/ir/task_results.py:33]: I don't think this is following the docstring format in #956

**Review by weinbe58 — APPROVED** [review]

**Comment by weinbe58** [comment]
This can be counted as a 16 USD bounty. Thank you for the contribution!

**Comment by WingCode** [comment]
@Roger-luo / @weinbe58
Could I be assigned to this issue please? So that the Unitary Hack bots can pick this up

**Comment by Roger-luo** [comment]
I will open separate issues for these merged PRs to make the bots work.

**Pull request merged**